### ISSUE TITLE
Sort outputs of gmt --help, --show-classic and --show-modules

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#3168]
The outputs are unsorted now.

**Comment by PaulWessel** [comment]
Not so easy sorting gmt --help.  Best if moduleinfo.h was sorted when created.  Can cmake add a sort step? Must be per suppl dir for the supplements.  If not then I can do it in C I think.

**Comment by seisman** [comment]
After #3174 was merged into master, it becomes more tricky to sort the output of `gmt --help`. 

The problem is that, we're using the gen_gmt_moduleinfo_h macro to build both gmt_core_moduleinfo.h and gmt_supplements_moduleinfo.h. The CMake sort function works for core modules, but not for supplemental modules. Using the sort functions gives:
```
===  GMT suppl: The official supplements to the Generic Mapping Tools  ===

Module name:     Purpose of spotter module:
----------------------------------------------------------------
backtracker      Generate forward and backward flowlines and hotspot tracks

Module name:     Purpose of seis module:
----------------------------------------------------------------
coupe            Plot cross-sections of focal mechanisms

Module name:     Purpose of geodesy module:
----------------------------------------------------------------
earthtide        Compute grids or time-series of solid Earth tides

Module name:     Purpose of potential module:
----------------------------------------------------------------
gmtflexure       Compute flexural deformation of 2-D loads, forces, and bending moments
gmtgravmag3d     Compute the gravity/magnetic anomaly of a 3-D body by the method of Okabe

Module name:     Purpose of spotter module:
----------------------------------------------------------------
gmtpmodeler      Evaluate a plate motion model at given locations
```

**Comment by PaulWessel** [comment]
Yes, the sorting has to be on a per subdir basis. Either we need a true/false flag to the macro to enable sorting or not (and then I do the sorting in C), or the macro somehow needs to sort for each subdir then append.

**Comment by PaulWessel** [comment]
Let me know if this needs to be done in the glue functions, @seisman.

**Comment by seisman** [comment]
Yes, better to sort that in the C codes.
